### PR TITLE
fix: Fix update datasource draft logic

### LINKED ISSUE #23681: [Bug]: TypeError in DatasourceSagas APPSMITH-6E1
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://appsmith.sentry.io/issues/4191020149

### Steps To Reproduce

NA

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### LINKED ISSUE #23686: [Bug]: TypeError in DatasourcesSagas APPSMITH-6D7
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://appsmith.sentry.io/issues/4183444471/

### Steps To Reproduce

NA

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by Irongade**
When update datasource draft saga is called for Rest APIs, an incorrect redux form name is used, hence the error.

Fixes #23686
Fixes #23681 

- Bug fix (non-breaking change which fixes an issue)

> Please describe the tests that you ran to verify your changes. Also list any relevant details for your test configuration.
> Delete anything that is not relevant
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/sagas/DatasourcesSagas.ts [hunk 1]
```diff
@@ -949,15 +949,14 @@ function* formValueChangeSaga(
       );
     }
   }
-  if (form !== DATASOURCE_DB_FORM && form !== DATASOURCE_REST_API_FORM) return;
+  if (!(form === DATASOURCE_DB_FORM || form === DATASOURCE_REST_API_FORM))
```
Comment (AmanAgarwal041) [inline app/client/src/sagas/DatasourcesSagas.ts:952]: Is this clearer now, which is why we made the change? @Irongade

Comment (Irongade) [reply app/client/src/sagas/DatasourcesSagas.ts:952]: The reason for this change is to make sure updatedraftsaga is called for either RestApiForm or DatasourceDBForm

Comment (AmanAgarwal041) [reply app/client/src/sagas/DatasourcesSagas.ts:952]: Both the conditions are same though.

Comment (Irongade) [reply app/client/src/sagas/DatasourcesSagas.ts:952]: Yes, I know. I wanted to go for conditions that were easier to read and understand but I guess I'll change it back.

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**